ci/Jenkinsfile

```diff
@@ -1,7 +1,30 @@
 node(label: 'linux') {
-  checkout scm
-  sh 'docker build -t ipfs-companion .'
-  sh 'docker run -i ipfs-companion yarn test'
-  sh 'docker run -i -v $(pwd)/build:/usr/src/app/build ipfs-companion yarn build'
-  archiveArtifacts artifacts: 'build/*.zip'
+    stage('git:checkout') {
+        checkout scm
+        sh 'printenv'
+    }
+    stage('ci:install') {
+        sh 'docker build -t ipfs-companion:${JOB_BASE_NAME}-${BUILD_NUMBER} .'
+    }
+    catchError {
+        stage('ci:test') {
+            sh 'docker run -i --name ipfs-companion-test-${JOB_BASE_NAME}-${BUILD_NUMBER} -e JUNIT_REPORT_PATH=test/report.xml ipfs-companion:${JOB_BASE_NAME}-${BUILD_NUMBER} npm run ci:test'
+            sh 'docker cp ipfs-companion-test-${JOB_BASE_NAME}-${BUILD_NUMBER}:/usr/src/app/test/report.xml test/report.xml || true' // copy test report
```

Comment:
`|| true'` is a workaround for false-negative bug in docker (more details @ https://github.com/moby/moby/issues/3986)